spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/JdbcPollingChannelAdapter.java

```diff
@@ -113,10 +115,10 @@ protected PreparedStatementCreator getPreparedStatementCreator(String sql,
 	 * @param rowMapper the {@link RowMapper} to use.
 	 */
 	public void setRowMapper(@Nullable RowMapper<?> rowMapper) {
-		this.rowMapper = rowMapper;
 		if (rowMapper == null) {
-			this.rowMapper = new ColumnMapRowMapper();
+			rowMapper = new ColumnMapRowMapper();
```

Comment:
To re-assign method argument is bad practice.
We should always treat them as `final`.
Please, consider `if..else` instead.